server/scripts/start-test-server.sh

```diff
@@ -48,12 +50,23 @@ check_containers_health() {
   all_healthy=true
 
   for container in $(docker ps -q); do
+    health_field=$(docker inspect --format='{{.State.Health}}' "$container")
+    if [ "$health_field" == "<nil>" ]; then
+      echo "Container has no health status. Skipped."
+      continue
+    fi
     name=$(docker inspect --format='{{.Name}}' "$container" | sed 's/\///')
+    app_name=$(docker inspect "$container" | jq -r '.[0].Config.Labels["com.docker.compose.project"]')
     health_status=$(docker inspect --format='{{.State.Health.Status}}' "$container")
 
+    if [ "$app_name" != "$APPLICATION" ]; then
```

Comment:
When two tests suites run simultaneously (auth and integration tests) - filter containers by application name to avoid check of unrelated containers.